### PR TITLE
feat: deploy Grafana using operator

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: components/monitoring/base/grafana-app.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    auth:
+      disable_login_form: False
+      disable_signout_menu: True
+    auth.basic:
+      enabled: False
+    auth.proxy:
+      enabled: True
+      enable_login_token: True
+      header_property: username
+      header_name: X-Forwarded-User
```
Comment (bamachrn) [inline components/monitoring/base/grafana-app.yaml:58]: How about allowing the users to have edit access to create their own dashboard?

File: components/monitoring/base/grafana-app.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      - name: OAUTH2_PROXY_CLIENT_SECRET
+        valueFrom:
+          secretKeyRef:
+            key: client-secret
+            name: grafana-oauth2-proxy-config
+      - name: OAUTH2_PROXY_COOKIE_SECRET
+        valueFrom:
+          secretKeyRef:
+            key: cookie-secret
+            name: grafana-oauth2-proxy-config
```
Comment (bamachrn) [inline components/monitoring/base/grafana-app.yaml:94]: Let's make sure we are adding this secret to README as well

**Comment by bamachrn** [comment]
@xcoulon  this needs clean up. seems like changes from multiple places came together

File: components/monitoring/base/configure-prometheus.yaml [hunk 3]
```diff
@@ -0,0 +1,223 @@
+---
```
Comment (xcoulon) [inline components/monitoring/base/configure-prometheus.yaml:1]: this file will be removed from this PR once https://github.com/redhat-appstudio/infra-deployments/pull/849 is merged (but I need it for now to test the connection to Prometheus)

**Comment by xcoulon** [comment]
> @xcoulon this needs clean up. seems like changes from multiple places came together

yes, I don't know what happened here, but it should be better now that I pushed https://github.com/redhat-appstudio/infra-deployments/commit/7a80a788e8479a721c6aab9ef52f096c2957586c 🤷‍♂️

File: components/monitoring/base/grafana-app.yaml (hunk 1 above)
Comment (xcoulon) [reply components/monitoring/base/grafana-app.yaml:51]: As already discussed on slack (IIRC), I don't think we need something like this for now.

File: components/monitoring/base/grafana-app.yaml (hunk 2 above)
Comment (xcoulon) [reply components/monitoring/base/grafana-app.yaml]: this should be covered by https://github.com/redhat-appstudio/infra-deployments/pull/883

**Review by amfred — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**